R/format_GLA.R

```diff
@@ -0,0 +1,688 @@
+                                                                    TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$ObservedHatch))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  Hatchlings = as.integer(.data$Hatchlings),
+                  Fledglings = as.integer(.data$Fledglings),
+                  MaleRing = as.character(.data$MaleRing),
+                  FemaleRing = as.character(.data$FemaleRing),
+                  ClutchSize = as.integer(.data$ClutchSize)) %>%
+
+    ## Arrange
+    dplyr::arrange(.data$PopID, .data$BreedingSeason, .data$LocationID, .data$FirstEggDate) %>%
+
+    ## Create additional variables that will be used in multiple data tables
+    dplyr::mutate(ClutchSize_max = dplyr::case_when(.data$ClutchComplete == "TRUE" ~ as.numeric(.data$ClutchSize),
+                                                    .data$ClutchComplete ==  "FALSE" ~ Inf),
+                  Species = dplyr::case_when(.data$Species == "greti" ~ species_codes[species_codes$SpeciesID == 14640,]$Species,
+                                             .data$Species == "bluti" ~ species_codes[species_codes$SpeciesID == 14620,]$Species,
+                                             .data$Species == "piefl" ~ species_codes[species_codes$SpeciesID == 13490,]$Species,
+                                             .data$Species == "nutha" ~ species_codes[species_codes$SpeciesID == 14790,]$Species,
+                                             .data$Species == "tresp" ~ species_codes[species_codes$SpeciesID == 15980,]$Species),
+                  BroodSize_observed = .data$ClutchSize - as.integer(.data$UnhatchedEggs),
+                  PopID = dplyr::case_when(.data$PopID == "cashel" ~ "CAS",
+                                           .data$PopID == "garscube" ~ "GAR",
+                                           .data$PopID == "kelvingrove_park" ~ "KEL",
+                                           .data$PopID == "sallochy" ~ "SAL",
+                                           .data$PopID == "SCENE" ~ "SCE",
+                                           TRUE ~ as.character(.data$PopID)),
+                  LayDate_observed = .data$FirstEggDate) %>%
+
+    ## Rename
+    dplyr::rename(FemaleID = .data$FemaleRing,
+                  MaleID = .data$MaleRing,
+                  LayDate_min = .data$FirstEggDate,
+                  LayDate_max = .data$LayingComplete,
+                  ClutchSize_observed = .data$ClutchSize,
+                  HatchDate_observed = .data$ObservedHatch,
+                  NumberFledged_observed = .data$Fledglings) %>%
+
+    ## Create BroodID based on PopID and row number
+    dplyr::ungroup() %>%
+    dplyr::mutate(BroodID = dplyr::case_when(!is.na(.data$Species) ~ paste(.data$PopID, dplyr::row_number(), sep ="-"))) %>%
```

Comment:
I would usually include info on BreedingSeason and NestBox, but this is maybe better because you don't need to worry about how to deal with multiple clutches in the same box/year. I would include this in the function description so it's easy for a user to decipher the BroodID values.